scaleway/resource_compute_instance_ip.go

```diff
@@ -121,6 +130,25 @@ func resourceScalewayComputeInstanceIPUpdate(d *schema.ResourceData, m interface
 		}
 	}
 
+	if d.HasChange("server_id") {
+		serverID := expandID(d.Get("server_id"))
+		if serverID != "" {
+			_, err = instanceApi.AttachIP(&instance.AttachIPRequest{
```

Comment:
What happened if you try to attach an IP that is already attached >

Reply:
detach/attach is done on the API side